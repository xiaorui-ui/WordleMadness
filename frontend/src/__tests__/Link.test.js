import renderer from 'react-test-renderer';
import Link from '../testfiles/Link';

it('changes the class when hovered', () => {
    const component = renderer.create(
        <Link page="http://localhost:3000/"> Welcome </Link>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseEnter();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        tree.props.onMouseLeave();
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});