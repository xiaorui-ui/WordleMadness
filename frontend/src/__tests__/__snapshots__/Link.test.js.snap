// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<a
  className="normal"
  href="http://localhost:3000/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   Welcome 
</a>
`;

exports[`changes the class when hovered 2`] = `
<a
  className="hovered"
  href="http://localhost:3000/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   Welcome 
</a>
`;

exports[`changes the class when hovered 3`] = `
<a
  className="normal"
  href="http://localhost:3000/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
   Welcome 
</a>
`;
